### ISSUE TITLE
Highligting text and Per element color picking. 

### ISSUE THREAD

**Issue by sohrabhamza** [zombieFox/nightTab#289]
**What kind of feature request is this?**

_Feature improvements – suggestions for something that can make a current feature a bit better._

I suggest that the ability to highlight text should be included. 
[Sample](https://imgur.com/khitOSC)

It adds to the aesthetic and generally makes it look better. 

Also the ability to change font colors per element would also be appreciated. 

Thanks!

**Comment by zombieFox** [comment]
Hi, text can be selected in nightTab. What are you not able to highlight?

Please also include these details:

 - OS: [e.g. iOS]
 - Browser [e.g. chrome, safari]
 - Version [e.g. 22]
 - nightTab version [e.g. 6.0.0]

**Comment by sohrabhamza** [comment]
Hi,
You misunderstood. I can highlight text. I was talking about the effect that the highlighting has and was wondering if it could be a feature to highlight text and have it stay that way. Please check the sample that I have attatched to see what I mean.

**Comment by metruzanca** [comment]
> highlight text and have it stay that way

As described it sounds like you want to be able to select text on the app via mouse and have that saved between sessions. This is out of scope for the project (and is quite complicated of a feature to add)

Adding a text background on the other hand is quite simple to do via css ([sample](https://i.imgur.com/u1K4xCz.png)) and could indeed potentially be cool if styled a bit better. ([Sample2](https://i.imgur.com/eQl38pe.png))

However, this opens up and interesting point. It could be pretty cool to allow custom css. (*this would be a big feature to add so don't get your hopes up and also might have performance implications*) This could easily allow custom font color per element, which in app could be quite tedious and make the theming system much more unnecessarily complex.

**Comment by sohrabhamza** [comment]
> As described it sounds like you want to be able to select text on the app via mouse and have that saved between sessions. This is out of scope for the project (and is quite complicated of a feature to add)

No, no this is not what I meant. I was just talking about the way highlighing made it look. 

Could there just not be a button to enable that says "add accent background to <specific element>"?

**Comment by zombieFox** [comment]
Hi, thanks for the suggestion. This is an interesting feature request. 

Firstly, some context -- one of the goals of the [v7.x](https://github.com/zombieFox/nightTab/releases/tag/v7.3.0) release was to greatly improve text colour contrast. This meant that all the text has to [know what colour is behind it](https://github.com/zombieFox/nightTab/blob/b2ceaffc87141ffb7144480b330cebc21deb93ea/src/component/clock/index.css#L45) to change to either black or white to be readable.

For this reason the project employs a lot of logic to check nearly all text whenever a setting is changed ([here is just one example](https://github.com/zombieFox/nightTab/blob/b2ceaffc87141ffb7144480b330cebc21deb93ea/src/component/toolbarControl/index.js#L241)). Especially the main layout background colour, (image backgrounds are a different story as I count them as out of the scope of nightTabs control since the user can add any image). 

I fear adding options to add custom colour behind the each part of the header would introduce more complexity and would be very difficult to develop. I'm sure the number of edge cases would greatly increase the possibility of text being unreadable due to poor contrast.

I'll mark this as investigate for now -- but I think this is might be an example of adding too many options which would lead too greater confusing code logic.

**Comment by sohrabhamza** [comment]
I get it from a complexity standpoint where you have to check against each color. 
But I think that if you just had an option to change the text color manually for each element, that it would handle all the "edge cases". If the user sees that the text is not visible, they can just manually change it themselves. 

Also I'm not sure if it'll work here buy check out [colorkt](https://github.com/kdrag0n/colorkt)

**Comment by metruzanca** [comment]
> Also I'm not sure if it'll work here buy check out colorkt. I'm saying this as it's used in a port of Android 12's material you which is doing something similar to what you said.

Thats for kotlin & java. Nighttab runs in the browser and so javascript is used.

> If the user sees that the text is not visible, they can just manually change it themselves.

Well nighttab does this automatically to provide a good user experience. Always beeing able to see text. The only case in which it is impossible to do so is if a background image is used, so giving the user the ability to change it themselves could indeed be a good idea.

**Comment by sohrabhamza** [comment]
> Thats for kotlin & java. Nighttab runs in the browser and so javascript is used.
Sorry for that. 

> so giving the user the ability to change it themselves could indeed be a good idea.
Hope to see it soon haha
